controllers/metal3.io/host_state_machine.go

```diff
@@ -226,6 +230,24 @@ func (hsm *hostStateMachine) checkInitiateDelete() bool {
 	return true
 }
 
+func (hsm *hostStateMachine) checkDetachedHost(info *reconcileInfo) (result actionResult) {
+	// If the detached annotation is set we remove any host from the
+	// provisioner and take no further action
+	// Note this doesn't change the current state, only the OperationalStatus
```

Comment:
We should ignore the annotation if the provisioning state is Unmanaged, to avoid changing the operational state from "discovered" to "detached". (Or, alternatively, change it back when the annotation is removed below.)

Where did we get to with the idea of only applying this in Provisioned/ExternallyProvisioned states?

Reply:
> Where did we get to with the idea of only applying this in Provisioned/ExternallyProvisioned states?

I think that would be fine for the use-case driving this change, we'd need to ensure it's also possible to delete the BMH from those states without triggering deprovisioning though

Reply:
The latest version restricts the API to only  [Externally]Provisioned)